test-suite/js/src/test/require-scala2/org/scalajs/testsuite/jsinterop/NonNativeJSTypeTestScala2.scala

```diff
@@ -46,4 +65,20 @@ object NonNativeJSTypeTestScala2 {
 
   class SomeValueClass(val i: Int) extends AnyVal
 
+  object ConstructorSuperCallWithDefaultParamsAndEarlyInitializers {
+    val sideEffects = mutable.ListBuffer.empty[String]
+
+    class Parent(parentParam1: Any = "param1", parentParam2: Any = "param2") extends js.Object {
```

Comment:
Use a default parameter that depends on another parameter? (IIRC this is allowed with multiple argument lists).

Reply:
Good point! Done.